content/token22-metadata.md

```diff
@@ -0,0 +1,1083 @@
+
+```ts
+  const createMintAccountInstructions = SystemProgram.createAccount({
+    fromPubkey: payer.publicKey,
+    lamports,
+    newAccountPubkey: mint.publicKey,
+    programId: TOKEN_2022_PROGRAM_ID,
+    space: totalLen,
+  });
+
+  const initMetadataPointerInstructions = createInitializeMetadataPointerInstruction(
+    mint.publicKey,
+    payer.publicKey,
+    mint.publicKey, // we will point to the mint it self as the metadata account
+    TOKEN_2022_PROGRAM_ID,
+  );
+
+  const initMintInstructions = createInitializeMintInstruction(
+    mint.publicKey,
+    decimals,
+    payer.publicKey,
+    payer.publicKey,
+    TOKEN_2022_PROGRAM_ID,
+  );
+
+  const initMetadataInstruction = createInitializeInstruction({
+    programId: TOKEN_2022_PROGRAM_ID,
+    mint: mint.publicKey,
+    metadata: mint.publicKey,
+    name: metadata.name,
+    symbol: metadata.symbol,
+    uri: metadata.uri,
+    mintAuthority: payer.publicKey,
+    updateAuthority: payer.publicKey,
+  });
+
+  const updateMetadataFieldInstructions = createUpdateFieldInstruction({
+    metadata: mint.publicKey,
+    updateAuthority: payer.publicKey,
+    programId: TOKEN_2022_PROGRAM_ID,
+    field: metadata.additionalMetadata[0][0],
+    value: metadata.additionalMetadata[0][1],
+  });
+  ```
+
+Wrap all of these instructions in a transaction to create the embedded NFT:
+```ts
+const transaction = new Transaction().add(
+  createMintAccountInstructions,
+  initMetadataPointerInstructions,
+  initMintInstructions,
+  initMetadataInstruction,
+  updateMetadataFieldInstructions, // if you want to add any custom field
+);
+const sig = await sendAndConfirmTransaction(connection, transaction, [payer, mint]);
+```
+
+Again, the order here matters.
+
+Note: The `createUpdateFieldInstruction` updates only one field at a time. If you want to have more than one custom field, you will have to call this method multiple times. Additionally, you can use the same method to update the basic metadata fields as well:
+
+```ts
+  const updateMetadataFieldInstructions = createUpdateFieldInstruction({
+    metadata: mint.publicKey,
+    updateAuthority: payer.publicKey,
+    programId: TOKEN_2022_PROGRAM_ID,
+    field: 'name', // Field | string
+    value: 'new name',
+  });
+```
+
+# Lab
+
+Now it is time to practice what we have learned so far. In this lab, we'll create a script that will illustrate how to create an NFT with the `metadata` and `metadata pointer` extensions.
+
+## 0. Getting started
+
+Let's go ahead and clone our starter code:
+```bash
+git clone https://github.com/Unboxed-Software/solana-lab-token22-metadata.git
+cd solana-lab-token22-metadata
+git checkout starter
+npm install
+```
+
+Let's take a look at what's been provided in the `starter` branch.
+
+Along with the NodeJS project being initialized with all of the needed dependencies, two other files have been provided in the `src/` directory.
```

Comment:
It would be better if you asked students to do this from scratch - just an `nom i` and esrun works for nearly everything else.